relay-event-schema/src/protocol/attributes.rs

```diff
@@ -236,6 +236,24 @@ impl Attributes {
         Some(&self.0.get(key)?.value()?.value.value)
     }
 
+    /// Returns the annotated attribute with the given key.
+    pub fn get_raw<Q>(&self, key: &Q) -> Option<&Annotated<Attribute>>
+    where
+        String: Borrow<Q>,
+        Q: Ord + ?Sized,
+    {
+        self.0.get(key)
+    }
+
+    /// Mutably returns the annotated attribute with the given key.
+    pub fn get_raw_mut<Q>(&mut self, key: &Q) -> Option<&mut Annotated<Attribute>>
+    where
+        String: Borrow<Q>,
+        Q: Ord + ?Sized,
+    {
+        self.0.get_mut(key)
+    }
+
```

Comment:
Don't have to change that now, but beginning to wonder if we shouldn't just make `self.0` public and if anyone needs raw access, they just access it raw. Otherwise we're just re-implementing the entire `BTreeMap` API surface.

We didn't do a `Deref` for a reason (clashes) but `.0` shouldn't be a problem.

Reply:
Yeah it's crossed my mind as well.